src/main/java/org/mitre/synthea/world/agents/Provider.java

```diff
@@ -235,13 +272,63 @@ public static void loadProviders(String state, String abbreviation, String filen
         if ("Yes".equals(row.remove("emergency"))) {
           parsed.servicesProvided.add(Provider.EMERGENCY);
         }
-
+        
         // add any remaining columns we didn't explicitly map to first-class fields
         // into the attributes table
         for (Map.Entry<String, String> e : row.entrySet()) {
           parsed.attributes.put(e.getKey(), e.getValue());
         }
         
+        String city = parsed.city;
+        String address = parsed.address;
+
+        //int population = (int) location1.getPopulation(city);
+        
+        //TODO - create a map of specialty:#
+        
+        boolean found = false;
+        //only enter this if THERE IS AN INFO FILE
+        //if there is no info file or if the provider is not found in the info file
+        // make them one general clinician that's always chosen
+        if (csvSpecialties == null || !found) {
+          parsed.specialtyInfo.put("GENERAL PRACTICE", 1);
+          parsed.clinicianMap.put("GENERAL PRACTICE", 
+              generateClinicianList(parsed, 1, "GENERAL PRACTICE"));
+        } else {
+          for (Map<String,String> rowSpecialty : csvSpecialties) {
+            if (rowSpecialty.get("address").equals(address) && rowSpecialty.get("city").equals(city)
```

Comment:
This is essentially going to do a "join" by address-city-state by looping over the entire file. It will work but it's slow. This is where combining the two files into one would help a lot.